File: dataset/cpp/test_header_image13.c
/*****************************************************************************
 * Test cases for libxlsxwriter.
 *
 * Test to compare output against Excel files.
 *
 * Copyright 2014-2022, John McNamara, jmcnamara@cpan.org
 *
 */

#include "xlsxwriter.h"

int main() {

    lxw_workbook  *workbook  = workbook_new("test_header_image13.xlsx");
    lxw_worksheet *worksheet = workbook_add_worksheet(workbook, NULL);

    lxw_header_footer_options header_options = {.image_left   = "images/black_72.jpg",
                                                .image_center = "images/black_150.jpg",
                                                .image_right  = "images/black_300.jpg"};

    worksheet_set_header_opt(worksheet, "&L&G&C&G&R&G", &header_options);

    return workbook_close(workbook);
}

File: dataset/python-mutated/test_header_image13.py
from ..excel_comparison_test import ExcelComparisonTest
from ...workbook import Workbook

class TestCompareXLSXFiles(ExcelComparisonTest):
    """
    Test file created by XlsxWriter against a file created by Excel.

    """

    def setUp(self):
        if False:
            return 10
        self.set_filename('header_image13.xlsx')
        self.ignore_elements = {'xl/worksheets/sheet1.xml': ['<pageMargins', '<pageSetup']}

    def test_create_file(self):
        if False:
            while True:
                i = 10
        'Test the creation of a simple XlsxWriter file with image(s).'
        workbook = Workbook(self.got_filename)
        worksheet = workbook.add_worksheet()
        worksheet.set_header('&L&G&C&G&R&G', {'image_left': self.image_dir + 'black_72.jpg', 'image_center': self.image_dir + 'black_150.jpg', 'image_right': self.image_dir + 'black_300.jpg'})
        workbook.close()
        self.assertExcelEqual()